source/jormungandr/jormungandr/interfaces/v1/serializer/base.py

```diff
@@ -213,25 +213,36 @@ def as_getter(self, serializer_field_name, serializer_cls):
         return lambda *args, **kwargs: self.value
 
 
-def flatten(obj):
-    if type(obj) != dict:
-        raise ValueError("Invalid argument")
-    new_obj = {}
-    for key, value in obj.items():
-        if type(value) == dict:
-            tmp = {'.'.join([key, _key]): _value for _key, _value in flatten(value).items()}
-            new_obj.update(tmp)
-        else:
-            new_obj[key] = value
-    return new_obj
+def value_by_path(obj, path, default=None):
+    """
+    >>> value_by_path({'a': {'b' : {'c': 42}}}, 'a.b.c')
+    42
+    >>> value_by_path({'a': {'b' : {'c': 42}}}, 'a.b.c.d', default=4242)
+    4242
+    >>> value_by_path({'a': {'b' : {'c': 42}}}, 'a.b.c.d', default=0)
+    0
+    >>> value_by_path({'a': {'b' : {'c': 0}}}, 'a.b.c', default=None)
+    0
```

Comment:
I'm not fan of testing `0 != None` here : @xlqian did you checked that test crashes without the "is not None" below ?